lib/msal-native-auth/src/UserAccountAttributes.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    InvalidAttributeErrorCode,
+    UserAccountAttributeError,
+} from "./error/UserAccountAttributeError.js";
+
+export class UserAccountAttributes {
+    private attributes: Record<string, string>;
+
+    constructor() {
+        this.attributes = {};
+    }
+
+    setCustomAttribute(name: string, value: string): void {
+        if (!name) {
+            throw new UserAccountAttributeError(
+                InvalidAttributeErrorCode,
+                name,
+                value
+            );
+        }
+
+        this.attributes[name] = value;
+    }
+
+    setCity(value: string): void {
+        this.attributes["city"] = value;
+    }
+
+    setCountry(value: string): void {
+        this.attributes["country"] = value;
+    }
+
+    setDisplayName(value: string): void {
+        this.attributes["displayName"] = value;
+    }
+
+    setGivenName(value: string): void {
+        this.attributes["givenName"] = value;
+    }
+
+    setJobTitle(value: string): void {
+        this.attributes["jobTitle"] = value;
+    }
+
+    setPostalCode(value: string): void {
+        this.attributes["postalCode"] = value;
+    }
+
+    setState(value: string): void {
+        this.attributes["state"] = value;
+    }
+
+    setStreetAddress(value: string): void {
+        this.attributes["streetAddress"] = value;
+    }
+
+    setSurname(value: string): void {
+        this.attributes["surname"] = value;
+    }
+
```

Comment:
These attributes are the default user profile attributes supported by Entra ID. With these set methods, it will be very handy for the users to set the attributes without knowing the attribute programmable name. https://learn.microsoft.com/en-us/entra/external-id/customers/concept-user-attributes#built-in-user-attributes